index.js

```diff
@@ -91,26 +101,39 @@ async function init() {
 		const parsedUrl = new URL(query.get('url'));
 		[, user, repository, ref, dir] = urlParserRegex.exec(parsedUrl.pathname);
 
-		console.log('Source:', {user, repository, ref, dir});
+		isGHE = parsedUrl.hostname !== 'github.com';
+		if (isGHE) {
+			domain = parsedUrl.hostname;
+			apiEndpoint = `${parsedUrl.protocol}//${parsedUrl.host}/api/v3`;
+
+			document.querySelector('#create-token').href = `${parsedUrl.protocol}//${parsedUrl.host}/settings/tokens/new?description=Download GitHub directory&scopes=repo`;
+		}
+
+		console.log('Source:', {domain, user, repository, ref, dir});
 	} catch {
 		return updateStatus();
 	}
 
+	const token = await waitForToken(domain);
+
 	if (!navigator.onLine) {
 		updateStatus('⚠ You are offline.');
 		throw new Error('You are offline');
 	}
 
 	updateStatus('Retrieving directory info…');
 
-	const {private: repoIsPrivate} = await fetchRepoInfo(`${user}/${repository}`);
+	const {private: repoIsPrivate} = await fetchRepoInfo(apiEndpoint, token, `${user}/${repository}`);
```

Comment:
This call is only needed to switch between `fetchPrivateFile` and `fetchPublicFile`. It needs to be changed to something like:

```suggestion
	const willDownloadViaAPI = isGHE || (await fetchRepoInfo(apiEndpoint, token, `${user}/${repository}`)).private;
```